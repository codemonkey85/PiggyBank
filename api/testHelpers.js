const testHelpers = {
  normalize: (text) => {
    return text.replace(/\s/g, "").toLowerCase();
  },
  mockRouter: () => {
    return {
      get: jest.fn(),
      post: jest.fn(),
      put: jest.fn(),
      delete: jest.fn()
    };
  },
  mockRepo: () => {
    return {
      account: {
        selectAll: jest.fn(),
        insert: jest.fn(),
        update: jest.fn(),
        delete: jest.fn()
      }
    };
  },
  mockRequest: () => {

  },
  mockResult: () => {
    return {
      json: jest.fn()
    };
  }
};

module.exports = testHelpers;