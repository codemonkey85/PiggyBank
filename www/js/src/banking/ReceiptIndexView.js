"use strict"

import { html, render } from "../../lib/lit-html/lit-html.js"

const template = (d) => html`
  <div class='row'>
    <div class='col'>
      <h1>Receipts</h1>
      <video playsinline autoplay style='width:340px;height:240px;'></video>
      <button id='snapshot-btn'>Take snapshot</button>
      <img id='snapshot-img' />
      <canvas></canvas>
    </div>
  </div>
`

const constraints = {
  audio: false,
  video: true
}

let imageCapture

export class ReceiptIndexView extends Backbone.View {
  render() {
    render(template(), this.el)

    navigator.mediaDevices
      .getUserMedia(constraints)
      .then(this.handleSuccess.bind(this))
      .catch(this.handleError)

    return this
  }

  // const video = document.querySelector('video');
  // canvas.width = video.videoWidth;
  // canvas.height = video.videoHeight;
  // canvas.getContext('2d').drawImage(video, 0, 0, canvas.width, canvas.height);
  // const canvas = window.canvas = document.querySelector('canvas');
  // canvas.width = 480;
  // canvas.height = 360;

  handleSuccess(stream) {
    window.stream = stream
    const video = this.$("video")[0]
    video.srcObject = stream
    const mediaStreamTrack = stream.getVideoTracks()[0]
    imageCapture = new ImageCapture(mediaStreamTrack)

    this.$("#snapshot-btn").on("click", this.takeSnapshot)
  }

  handleError(error) {
    console.log("navigator.MediaDevices.getUserMedia error: ", error.message, error.name)
  }

  
  takeSnapshot(e) {
    e.preventDefault()
    const img = document.querySelector("#snapshot-img")
    imageCapture.takePhoto()
      .then(blob => {
        img.src = URL.createObjectURL(blob);
        img.onload = () => { URL.revokeObjectURL(this.src); }
      })
      .catch(error => console.error("takePhoto() error: ", error))
  }

}
