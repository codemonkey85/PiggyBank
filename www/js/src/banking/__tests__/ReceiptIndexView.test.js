import { mockEvent } from "../../__tests__/testHelpers"
import { ReceiptIndexView } from "../ReceiptIndexView"

let view

beforeEach(() => {
  view = new ReceiptIndexView()
})

test("ReceiptIndexView has render method", () => {
  const renderedView = view.render()
  expect(view).toBe(renderedView)
  expect(view.el).toMatchSnapshot()
})

