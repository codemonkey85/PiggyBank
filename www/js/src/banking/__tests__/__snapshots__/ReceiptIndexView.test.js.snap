// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptIndexView.render() get media permissions 1`] = `
<div>
  <!---->
  
  
  <div
    class="row"
  >
    
    
    <div
      class="col"
    >
      
      
      <h1>
        Receipts
      </h1>
      
      
      <video
        autoplay=""
        playsinline=""
        style="width:340px;height:240px;"
      />
      
      
      <button
        id="snapshot-btn"
      >
        Take snapshot
      </button>
      
      
      <img
        id="snapshot-img"
      />
      
      
      <canvas />
      
    
    </div>
    
  
  </div>
  

  <!---->
</div>
`;
